module/product/src/Infrastructure/Filter/BatchAction/ProductDeleteBatchActionFilter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Product\Infrastructure\Filter\BatchAction;
+
+use Ergonode\BatchAction\Domain\ValueObject\BatchActionFilter;
+use Ergonode\BatchAction\Domain\ValueObject\BatchActionType;
+use Ergonode\BatchAction\Infrastructure\Provider\BatchActionFilterIdsInterface;
+use Ergonode\Product\Domain\Query\ProductQueryInterface;
+use Ergonode\SharedKernel\Domain\Aggregate\ProductId;
+use Ergonode\SharedKernel\Domain\AggregateId;
+
+class ProductDeleteBatchActionFilter implements BatchActionFilterIdsInterface
```

Comment:
Wondering wouldn't be proper to create a filter for products not product delete specific? I.e for edit and delete those may work exactly the same I think

Reply:
It could be hard because the type is `BatchActionType` defines as `supports`. 

I'll separate this into an abstract class

Reply:
Could just have array of types - up 2 u